File: backend/src/main/java/com/cdac/farmermarketplace/service/AdminUserService.java
package com.cdac.farmermarketplace.service;

import java.util.List;

import com.cdac.farmermarketplace.dto.response.AdminUserResponseDto;
import com.cdac.farmermarketplace.entity.Role;

public interface AdminUserService {

    // Get all users / search / filter
    List<AdminUserResponseDto> getUsers(
            String search,
            Role role
    );

    // Get single user
    AdminUserResponseDto getUserById(Long id);

    // Block user
    AdminUserResponseDto blockUser(Long id);

    // Unblock user
    AdminUserResponseDto unblockUser(Long id);
}

File: frontend/src/services/adminUserService.js
import api from "./api";

// Get all users / search / filter
export const getUsers = async (search = "", role = "") => {
  const params = {};

  if (search.trim()) {
    params.search = search.trim();
  }

  if (role) {
    params.role = role;
  }

  const response = await api.get("/admin/users", {
    params,
  });

  return response.data;
};

// Get user by ID
export const getUserById = async (id) => {
  const response = await api.get(`/admin/users/${id}`);
  return response.data;
};

// Block user
export const blockUser = async (id) => {
  const response = await api.patch(
    `/admin/users/${id}/block`
  );

  return response.data;
};

// Unblock user
export const unblockUser = async (id) => {
  const response = await api.patch(
    `/admin/users/${id}/unblock`
  );

  return response.data;
};